### PR TITLE
Seeds

### PR DESCRIPTION
**Pull request by breakfireaus**

### CODE REVIEW

**Comment by breakfireaus** [comment]
will be working on seeds tomorrow between 5 and 6:30 seeds should be completed, something unexpected just happened

**Comment by Auralise** [comment]
No problems. I will wait to review/approve until the work has continued.

**Comment by breakfireaus** [comment]
Can you please advise how to put date and time in for seeds date for service start and end @Auralise

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by Auralise** [comment]
I assume that it is the ISO standard which would be implemented in the mysql database. Id recommend checking out docs on this: https://www.w3schools.com/sql/sql_dates.asp

The Sequelize DataType.DATE maps to the datetime format in SQL. 

Frankly, I've never done it with sequelize so just play with it until it works

**Comment by breakfireaus** [comment]
So if I put the format as it is in w3schools it should be fine?



Sent from my Galaxy



-------- Original message --------
From: Auralise ***@***.***>
Date: 24/11/22 12:29 pm (GMT+10:00)
To: breakfireaus/Inter-book ***@***.***>
Cc: Matthew Younger-Chudnovsky ***@***.***>, Author ***@***.***>
Subject: Re: [breakfireaus/Inter-book] Seeds (PR #33)


I assume that it is the ISO standard which would be implemented in the mysql database. Id recommend checking out docs on this: https://www.w3schools.com/sql/sql_dates.asp<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.w3schools.com%2Fsql%2Fsql_dates.asp&data=05%7C01%7C%7Cfef9e92c06b44228dca308dacdbb4f61%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638048501641526969%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=bPHcD2BGP3XvKFiELn8Q6Jo5bjsul5oO1PbEnZhTRYk%3D&reserved=0>

The Sequelize DataType.DATE maps to the datetime format in SQL.

Frankly, I've never done it with sequelize so just play with it until it works

**Comment by Auralise** [comment]
I don't know, its seemingly a good starting point. I'd just try it and if it doesn't work, play with it and then just test it until it works

**Comment by breakfireaus** [comment]
Ok

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

File: seeds/industry.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const { Industry } = require('../models');
+
+const industryData = [
+  {
+    id: 1,
+    Industry: 'Dance',
```
Comment (Auralise) [inline seeds/industry.js]:
```suggestion
    industry: 'Dance',
```

File: seeds/industry.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  },
+
+  {
+    id: 2,
+    Industry: 'Construction',
```
Comment (Auralise) [inline seeds/industry.js]:
```suggestion
    industry: 'Construction',
```

File: seeds/industry.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  },
+
+  {
+    id: 3,
+    Industry: 'IT',
```
Comment (Auralise) [inline seeds/industry.js]:
```suggestion
    industry: 'IT',
```

File: seeds/industry.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  },
+
+  {
+    id: 4,
+    Industry: 'Fitness',
```
Comment (Auralise) [inline seeds/industry.js]:
```suggestion
    industry: 'Fitness',
```

File: seeds/industry.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  },
+
+  {
+    id: 5,
+    Industry: 'Law',
```
Comment (Auralise) [inline seeds/industry.js]:
```suggestion
    industry: 'Law',
```

File: seeds/service.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
+const { Service } = require('../models');
+
+const serviceData = [
+  {
+    id: 1,
+    title: 'Salsa Dance class',
+    start: '2022-12-05T12:00:000Z',
```
Comment (Auralise) [inline seeds/service.js]: These dates did not insert into the database successfully, producing the following error: 
image


As discussed, can you please experiment with datetime formats as specified for mysql and described on w3schools here: https://www.w3schools.com/sql/sql_dates.asp

Please test these next time before submitting for the next review as discussed.

File: seeds/userIndustry.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+    {
+        user_id: 3,
+        industry_id: 1
+    },
+
+    {
+        
+        user_id: 4,
+        industry_id: 6
```
Comment (Auralise) [inline seeds/userIndustry.js]:
```suggestion
        industry_id: 4 
```

This is an invalid industry ID of 6 as we have only defined the seeds for 5 industries.

**Review by Auralise — CHANGES_REQUESTED** [review]
Matt, please be sure to test these things before submitting them for review by simply running them with the debugger. The assumption will be, especially in the professional world, that code submitted for review has been tested and is working, at least for you and is at the stage where it is working in the larger context of the application. This is true at least of code that is to be merged into the main branch.

Granted this can be a bit different at the start of the project where not everything has been put in place to actually test within context of the application and frankly, we have done our project a bit backwards in this regard but as we have the sequelize models, we can actually test this against them to verify that they work. 

Please review the changes I have proposed and perform some testing of your own to verify that all of these seed files successfully insert into the database before requesting the next review. 

Additionally, you have submitted some other changes to the page routes page as part of this branch which is unrelated. Please try not to do this so that we can have branches that are contained to the issue they represent. I know this is easy to do but do keep an eye on it 

Thanks mate!

**Comment by breakfireaus** [comment]
@Auralise @asish-dewan all fixed

**Review by Auralise — APPROVED** [review]
I have tested the seeds and the database seems to be in the correct state after running the seed. Good job! 

I will merge this in now.

**Pull request merged**